detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/parse/GoModuleDependencyHelper.java

```diff
@@ -1,87 +1,83 @@
+        /* Correct lines that get mis-interpreted as a direct dependency, given the list of direct deps, requirements graph etc.*/
         for (String grphLine : graph) {
-            boolean containsDirect = false;
-            for (String directMod : directs) {
-                if (grphLine.startsWith(main) && grphLine.contains(directMod)) {
-                    containsDirect = true;
-                }
-            }
+            boolean containsDirect = containsDirectDependencies(directs, main, grphLine);
             
-            // splitting here allows matching with less effort
+            // Splitting here allows matching with less effort
             String[] splitLine = grphLine.split(" ");
-            boolean needsRedux = false;
-            if (!containsDirect && splitLine[0].equals(main)) {// anything that falls in here isn't a direct dependency of main
-                needsRedux = true;
-            }
-
-            // this searches for instances where the main module is apparently referring to itself.  This can
-            // step on the indirect dependency making it seem to be direct.
+            
+            // anything that falls in here isn't a direct dependency of main
+            boolean needsRedux = !containsDirect && splitLine[0].equals(main);
+            
+            /* This searches for instances where the main module is apparently referring to itself.  
+            This can step on the indirect dependency making it seem to be direct.*/
             if (splitLine[0].startsWith(main) && splitLine[0].contains("@")) {
-                boolean gotonext = false;
-                for (String adep : correctedDependencies) {
-                    if (splitLine[1].startsWith(adep)) {
-                        gotonext = true;
-                        break;
-                    }
-                }
-                if (gotonext) {
+                boolean gotoNext = hasDependency(correctedDependencies, splitLine[1]);
+                if (gotoNext) {
                     continue;
                 }
             }
 
             if (needsRedux) {
-                // Redo the line to establish the direct reference module to this *indirect* module
+                /* Redo the line to establish the direct reference module to this *indirect* module*/
                 grphLine = this.getProperParentage(grphLine, splitLine, whyMap, directs, correctedDependencies);
             }
-
-            if (!goModGraph.contains(grphLine)) {
-                logger.debug(grphLine);
-                goModGraph.add(grphLine);
-            }
+            
+            goModGraph.add(grphLine);
         }
         return goModGraph;
     }
+    
+    private boolean containsDirectDependencies(List<String> directs, String main, String grphLine) {
+        for (String directMod : directs) {
+            return grphLine.startsWith(main) && grphLine.contains(directMod);
+        }
+        return false;
+    }
```

Comment:
Using a loop to use first element and return was a bit unreadable for me. Let me know if my suggested refactor makes it easier!
```suggestion
    private boolean containsDirectDependencies(List<String> directs, String main, String grphLine) {
        String directMod = directs.get(0);
        return grphLine.startsWith(main) && grphLine.contains(directMod);
    }
```

Reply:
The first element check was a possible functional change which I want to avoid in this PR. But your point is valid - this can be made more readable.